astropy/io/fits/hdu/hdulist.py

```diff
@@ -689,6 +698,13 @@ def writeto(self, fileobj, output_verify='exception', clobber=False,
             hdu._prewriteto(checksum=checksum)
             try:
                 hdu._writeto(hdulist._file)
+            except IOError, e:
```

Comment:
The Python 3-compatible syntax for this is `except IOError as e:`  (although preferably use something like `exc` instead, just to avoid single-letter variable names)